File: C_Codes/array_operations.c
#include <stdio.h>
int main()
{
    int s, d, n;
    int arr[10] = {1, 3, 2, 4, 5, 15, 7, 78, 33, 43};

label:
    printf("\n\nPress a number form 1 to 5 for the following operations: \n Press 1 to search an element in array\n Press 2 for insertinng an element in the array\n");
    printf(" Press 3 for Deleting the element of the array\n Press 4 for Reversing an array\n Press 5 for updating the array\n");
    printf("\n Press 6 to return to menu\n Press 7 to Exit the program: ");
    scanf("%d", &n);

    switch (n)
    {
    case 1:
        printf("Enter the element to be searched: ");
        scanf("%d", &s);
        for (int i = 0; i < 10; i++)
        {
            if (arr[i] == s)
            {
                printf("The element is at %d indices", i);
            }
        }
        break;
    case 2:
        printf("\nEnter the position at which number is to be inserted: ");
        scanf("%d", &n);

        for (int i = 10 + 1; i >= n; i--)
        {
            {
                arr[i] = arr[i - 1];
            }
        }

        printf("\nEnter the element to be inserted: ");
        scanf("%d", &arr[n]);

        for (int i = 0; i < 10 + 1; i++)
        {
            printf("%d ", arr[i]);
        }
        break;
    case 3:
        printf("\n\nEnter the position of element to be deleted: ");
        scanf("%d", &d);

        for (int i = d; i < 10; i++)
        {
            arr[i] = arr[i + 1];
        }
        for (int i = 0; i < 10 - 1; i++)
        {
            printf("%d ", arr[i]);
        }
        break;
    case 4:
        for (int i = 0; i < 10 / 2; i++)
        {
            int b;
            b = arr[i];
            arr[i] = arr[10 - i - 1];
            arr[10 - i - 1] = b;
        }
        printf("\nThe reversed array is: ");
        for (int i = 0; i < 10; i++)
        {
            printf("%d ", arr[i]);
        }
        break;
    case 5:
        for (int i = 0; i < 10; i++)
        {
            if (i % 2 != 0)
            {
                arr[i] = arr[i] * 2;
            }
            else
            {
                arr[i] = arr[i] + 5;
            }
        }
        printf("\nThe updated array is: ");
        for (int i = 0; i < 10; i++)
        {
            printf("%d", arr[i]);
        }
        break;
    case 6:
        goto label;
    case 7:
        goto exit;
    default:
        break;
    }
exit:
    return 0;
}

File: Java_Codes/array_operations.java
// Write menu-driven program to input 1D array and perform following operations:
// a.Linear search for an element.
// b.Inserting an element at specified position.
// c.Deleting an element from the specified position.
// d.Reversing the array.
// e.Updating the array , multiply the odd-indexed elements by 2 and adding 5 to all the elements at even index.

import java.util.*;

public class array_operations {

    int linear_search(int array[], int ele, int length) {
        for (int i = 0; i < length; i++) {
            if (array[i] == ele) {
                return i;
            }
        }
        return -1;
    }

    int array_inserion(int array[], int ele, int place, int length) {

        for (int i = length + 1; i > place; i--) {
            array[i] = array[i - 1];
        }

        array[place] = ele;

        for (int i = 0; i < length + 1; i++) {
            System.out.print(array[i] + " ");
        }
        return length = length + 1;
    }

    int array_deletion(int array[], int ele, int length) {
        int result = linear_search(array, ele, length);
        if (result == -1) {
            System.out.println("The element to be deleted is not in the array.");
        } else {
            for (int i = result; i < length; i++) {
                array[i] = array[i + 1];
            }

            System.out.print("The updated array after deletion of element is: ");
            for (int i = 0; i < length - 1; i++) {
                System.out.print(array[i] + " ");
            }
        }
        return length = length - 1;
    }

    void array_reverse(int array[], int length) {
        int temp;
        for (int i = 0; i < length / 2; i++) {
            temp = array[i];
            array[i] = array[length - 1 - i];
            array[length - 1 - i] = temp;
        }
        System.out.print("The reversed array is: ");
        for (int i = 0; i < length; i++) {
            System.out.print(array[i] + " ");
        }
    }

    void array_updation(int array[], int length) {
        for (int i = 0; i < length; i++) {
            if (i % 2 == 0) {
                array[i] = array[i] * 2;
            } else {
                array[i] = array[i] + 5;
            }
        }
        System.out.print("The updated array is: ");
        for (int i = 0; i < length; i++) {
            System.out.print(array[i] + " ");
        }
    }

    void array_sorting(int array[], int length) {
        int min;
        for (int i = 0; i < length; i++) {
            for (int j = i + 1; j < length; j++) {
                if (array[j] < array[i]) {
                    min = array[j];
                    array[j] = array[i];
                    array[i] = min;
                }
            }
        }

        System.out.println("The sorted array is: ");
        for (int i = 0; i < length; i++) {
            System.out.print(array[i] + " ");
        }
    }

    public static void main(String[] args) {
        int size, key, num, pos, result;
        Scanner scr = new Scanner(System.in);
        array_operations func = new array_operations();

        int[] myArray = new int[100];

        System.out.print("Enter the size of the array less than 100: ");
        size = scr.nextInt();

        System.out.println("Enter the elements of the array: ");
        for (int i = 0; i < size; i++) {
            myArray[i] = scr.nextInt();
        }

        System.out.print("The enetered array is: ");
        for (int i = 0; i < size; i++) {
            System.out.print(myArray[i] + " ");
        }

        label: while (true) {
            System.out.println("\nPress 1 to 6 for the following array operations:");
            System.out.println(" 1 for linear search for an element.\n 2 for Inserion.\n 3 for Deletion.");
            System.out.println(" 4 for Reversing the array.\n 5 for Updation.");
            System.out.print(" 6 for sorting the array.\n 7 for exiting the program:");
            key = scr.nextInt();

            switch (key) {
                case 1:
                    System.out.print("\nEnter the element to be searched in the array: ");
                    num = scr.nextInt();

                    result = func.linear_search(myArray, num, size);

                    if (result == -1) {
                        System.out.println("The element is not in the array.");
                    } else {
                        System.out.println("The element is at the " + result + "th indices of the array.");
                    }
                    break;
                case 2:
                    System.out.print("Enter the position at which element is to be inserted: ");
                    pos = scr.nextInt();

                    System.out.print("Enter the element to be inserted: ");
                    num = scr.nextInt();

                    size = func.array_inserion(myArray, num, pos, size);
                    break;
                case 3:
                    System.out.print("Enter the element to be deleted: ");
                    num = scr.nextInt();

                    size = func.array_deletion(myArray, num, size);
                    break;
                case 4:
                    func.array_reverse(myArray, size);
                    break;
                case 5:
                    func.array_updation(myArray, size);
                    break;
                case 6:
                    func.array_sorting(myArray, size);
                    break;
                case 7:
                    break label;
                default:
                    break;
            }
        }
        scr.close();
    }
}